intelmq/bots/parsers/alienvault/parser_otx.py

```diff
@@ -27,16 +28,32 @@ def process(self):
 
         for pulse in json.loads(raw_report):
             additional = {"author": pulse['author_name'], "pulse": pulse['name']}
+
             for indicator in pulse["indicators"]:
                 event = self.new_event(report)
                 # hashes
                 if indicator["type"] in HASHES.keys():
                     event.add('malware.hash', HASHES[indicator["type"]] +
                               indicator["indicator"])
+                    additional['malware.hash.type'] = indicator["type"]
+                    additional['malware.hash.raw'] = indicator["indicator"]
                 # fqdn
-                if indicator["type"] in ['hostname', 'domain']:
-                    event.add('source.fqdn',
-                              indicator["indicator"])
+                elif indicator["type"] in ['hostname', 'domain']:
+                    # not all domains in the report are just domains
+                    # some are urls, we can manage those here instead
+                    # of raising errors
+                    #
+                    # dirty check if there is a scheme
+
+                    resource = indicator["indicator"] \
+                        if 'tp://' in indicator["indicator"] \
```

Comment:
missing https was an oversight. As for using startswith, i would say lets use "://" in check to make it more generic. Thoughts? I pushed the change